dcpy/extract/ingest.py

```diff
@@ -84,10 +90,92 @@ def extract_and_archive_raw_dataset(dataset: str, version: str | None):
     return config
 
 
-def transform_to_parquet(config: recipes.ExtractConfig):
-    """Given config of archived raw dataset, transform it to parquet"""
-    ### idea is that this will dump to PARQUET_PATH - other functions will assume parquet file is there as well
-    raise NotImplemented
+def transform_to_parquet(
+    config: recipes.ExtractConfig, local_data_path: Path | None = None
+):
+    """
+    Transforms raw data into a parquet file format and saves it locally.
+
+    This function first checks for the presence of raw data locally at the specified `local_data_path`.
+    If the path not provided, it is downloaded from an S3 bucket using the `config` parameter.
+    The raw data is then read into a GeoDataFrame and saved as a parquet file.
+
+    The transformation process varies depending on the format of the raw data, which can be in .shp, .gdb,
+    or .csv format. For csv files, if geometry is present, it is converted into a GeoSeries before creating
+    the GeoDataFrame.
+
+    Parameters:
+        config (recipes.ExtractConfig): Config object containing geometry info.
+        local_data_path (Path, optional): Path to the local data file. If not provided, data is pulled from S3 bucket.
+
+    Raises:
+        AssertionError: If `local_data_path` is provided but does not point to a valid file.
+    """
+
+    # create new dir for raw data and output parquet file
+    if TMP_DIR.is_dir():
+        shutil.rmtree(TMP_DIR)
+    TMP_DIR.mkdir()
+
+    if local_data_path:
+        # TODO: fix logic for cases when it's a directory, not a file
+        assert local_data_path.is_file(), "Local path should be a valid file"
+        logger.info(f"Raw data was found locally at {local_data_path}")
+    else:
+        local_data_path = TMP_DIR / config.raw_filename
+
+        s3.download_file(
+            bucket=recipes.BUCKET,
+            key=config.raw_dataset_s3_filepath,
+            path=local_data_path,
+        )
+        logger.info(f"Downloaded raw data from s3 to {local_data_path}")
+
+    data_load_config = config.transform_to_parquet_metadata
+
+    match data_load_config.format:
+        case "shapefile" | "geodabase":
+            gdf = gpd.read_file(
+                local_data_path,
+                geometry=data_load_config.geometry.geom_column,
+                crs=data_load_config.geometry.crs,
+                layer=data_load_config.geometry.layer,
+            )
+
+        case "csv":
+            df = pd.read_csv(
+                local_data_path,
+                index_col=False,
+                encoding=data_load_config.encoding,
+                delimiter=data_load_config.delimiter,
+            )
+
+            if (
+                not data_load_config.geometry.crs
+                and not data_load_config.geometry.geom_column
+            ):
+                gdf = df
+
+            else:
+                geom_column = data_load_config.geometry.geom_column
```

Comment:
Would be good to have validation that this column exists in the df

Reply:
are you thinking to have validation in code here or in a data contract?

Reply:
To me that sort of validation belongs here - if we're explicitly providing geospatial information as part of this conversion, I'd prefer a helpful error at time of conversion.

As written now, I think we'd simply get a pandas error. Which is relatively helpful, but my gut response is that it'd be helpful to see something like "specified geometry column does not exist in test.csv" or something like that